ring/ring.go

```diff
@@ -162,6 +163,7 @@ func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	cfg.KVStore.RegisterFlagsWithPrefix(prefix, "collectors/", f)
 
 	f.DurationVar(&cfg.HeartbeatTimeout, prefix+"ring.heartbeat-timeout", time.Minute, "The heartbeat timeout after which ingesters are skipped for reads/writes. 0 = never (timeout disabled).")
+	f.DurationVar(&cfg.UpdateInterval, prefix+"ring.update-interval", 0, "How often to recompute ring state when a change is detected from the KVStore. 0 = no delay.")
```

Comment:
I'd suggest mentioning that nonzero interval means rate-limiting the updates, which can have positive effect on cpu usage.